feature/58_SORTING_FUNCTIONALITY

Symphonie234 opened the pull request:
### Changes Made:

- Created a new method in the **`Listing`** **Model** for sorting listings by the most frequently associated tags.
- Added code in the **`HomeController`** which sorts listings based on a 'sort' query parameter, prioritizing options such as the latest, top-rated by likes, or defaulting to sorting by the most frequently used tag.
- Adjusted the code in the **`home/index.blade.php`** for the "Latest, Relevant, and Top" sorting to work.
- Added a new **JavaScript** file named **`sorting.js`** for handling sorting of listings.

Symphonie234 committed:
added a functionality for sorting the listings based on the relevance, latest, and top type

Symphonie234 committed:
switched pagination from 5 to 10

Symphonie234 commented:
### Additional Changes Made:

- Changed the pagination for listings in the **`HomeController`** **index** function from 5 to 10.

ChronoDevs commented on app/Http/Controllers/HomeController.php:
```diff
@@ -28,8 +23,22 @@ public function index()
+
+        // Sort the listings based on the 'sort' query parameter
+        $sort = $request->query('sort');
+        if ($sort === 'latest') {
```
make an enum and your queries please transfer to the model

ChronoDevs commented on app/Http/Controllers/HomeController.php:
```diff
@@ -28,8 +23,22 @@ public function index()
             ->take(10) // Limit to the top 10 tags
```
set limit from const file or enum

ChronoDevs commented on app/Http/Controllers/HomeController.php:
```diff
@@ -9,12 +9,7 @@
-     * @return \Illuminate\Contracts\Support\Renderable
-     */
-    public function index()
+    public function index(Request $request)
```
add comment method above

Symphonie234 committed:
Attended to the requested changes in the PR No. 1

Symphonie234 committed:
Merge branch 'develop' into feature/58_SORTING_FUNCTIONALITY

Symphonie234 commented:
### Changes Addressed in the PR No. 1:

- Made a new **Enum** file named **`ListingSort.php`**.
- Modified the **`HomeController`** to shorten it.
- Added two new functions in the **`Listing`** **Model** named **`sortBy`** and **`mostUsedTags`**
